qiskit/circuit/library/n_local/n_local.py

```diff
@@ -60,6 +60,70 @@ class NLocal(BlueprintCircuit):
 
     If specified, barriers can be inserted in between every block.
     If an initial state object is provided, it is added in front of the NLocal.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            # Example 1:
+            #   "Blocks" are defined by `QuantumCircuit` objects.
+            #   The "basic rotation block" consists of a 1-depth 2-qubits `QuantumCircuit` object.
+            #   The "basic entanglement block" consists of a 3-depth 4-qubits `QuantumCircuit` object.
+            #   Every "rotation layer" consists of a vertical repetition over the basic rotation block.
+            #   Every "entanglement layer" consists of a `linear` repetition over
+            #   the basic entanglement block.
+
+            from qiskit import QuantumCircuit
+            from qiskit.circuit import ParameterVector
+            from qiskit.circuit.library import NLocal
+
+            num_qubits = 6
+
+            # Basic rotation block pre-setting
+            num_qubits_rot_block = 2
+            theta = ParameterVector(name='theta', length=num_qubits_rot_block)
+            rot_block = QuantumCircuit(num_qubits_rot_block)
+            rot_block.ry(theta[0], 0)
+            rot_block.rx(theta[1], 1)
+
+            # Basic entanglement block pre-setting
+            num_qubits_ent_block = 4
+            ent_block = QuantumCircuit(num_qubits_ent_block)
+            for i in range(num_qubits_ent_block - 1):
+                ent_block.cx(i, i + 1)
+
+            # Assembling the pre-defined blocks into layers that form an `NLocal` circuit
+            nloc = NLocal(num_qubits, rotation_blocks=rot_block, entanglement_blocks=ent_block,
+                        entanglement='linear', insert_barriers=True)
+
+            display(nloc.decompose().draw(output='mpl'))
+
+        .. jupyter-execute::
+
+            # Example 2:
+            #   "Blocks" are defined directly by lists of (inherited) `Gate` objects.
+            #   The "basic rotation block" consists of an RY gate followed by an RX.
+            #   The "basic entanglement block" consists of a single CZ gate.
+            #   Every "rotation layer" consists of a vertical repetition over the basic rotation block.
+            #   Every "entanglement layer" consists of a `circular` repetition over
+            #   the basic entanglement block.
+
+
+            from qiskit.circuit import Parameter, ParameterVector
+            from qiskit.circuit.library import NLocal, RYGate, RXGate, CZGate
+
+            num_qubits = 4
+            theta = Parameter('theta')
+
+            # In the following `***_blocks` arguments is necessary to specify the actual gate objects,
+            # rather than some abbreviations that are used in other classes,
+            # e.g `CZGate()` = right, `cz` = wrong.
```

Comment:
If users are not familiar with specification per string, I think this is more confusing than clarifying. How about just removing that subsentence? 🙂 
```suggestion
            # In the following `***_blocks` arguments is necessary to specify the Gate objects
```

Reply:
Sounds OK, Removing.